### DISCUSSION TITLE
add subtitle/badge to menu icon when not expanded

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#198] [Ideas]
using this in card_mod
```
      .menu ha-icon-button {
        color: {{'var(--alert-color)' if alerts else 'var(--ok-color)'}} !important;
      }

      :host(:not([expanded])) .menu ha-icon-button:after {
        content: {%- set m_alerts = states('sensor.marquee_alerts')|int(0) %} {{m_alerts if m_alerts > 0 }};
        {{style_badge_not_expanded}}
        display: block;
        color: ivory;
        background: var(--alert-color);
        bottom: 20px;
      }
```

gave [me the number on the icon when not expanded.](https://community.home-assistant.io/t/card-mod-super-charge-your-themes/212176/1854?u=mariusthvdb)

the latest custom-sidebar release offers no way to set a notification on the menu icon in the config?

please consider adding an option to the non expanded sidebar menu icon

<img width="60" alt="Scherm­afbeelding 2024-11-12 om 09 23 58" src="https://github.com/user-attachments/assets/e285c818-63dc-435c-a326-afd43abe3b35">

**Comment by elchininet** [comment]
How does that work when the native notification is shown? Do you hide the native one or is it automatically replaced?

![image](https://github.com/user-attachments/assets/29a0fd3e-512f-4639-8f01-f31e48e8077b)

**Reply by elchininet** [reply]
I have checked, it is an extra element.

<img width="1411" alt="image" src="https://github.com/user-attachments/assets/0c534c5a-d578-4d6a-826e-e3aa96a0137d">

This is too specific, it should work only when the sidebar is not expanded and it should hide a native Home Assistant functionality (show a dot when there is a notification) to show something else in its place. I need to think about it to see if it can be achieved with a generic option.

**Comment by Mariusthvdb** [comment]
Uhm, what native notification is that? I’ve never seen anything there?


that is to say, in the companion app, there is no collapsed menu at all, (so also no notifications there), and, in Desktop, there is no badge on the notification tray in collapsed mode:

<img width="55" alt="Scherm­afbeelding 2024-11-12 om 15 01 03" src="https://github.com/user-attachments/assets/e680423d-d487-4712-97f9-7098d2f1a5d9">

with that badge set in card_mod I get

non collapsed (expanded)

<img width="251" alt="Scherm­afbeelding 2024-11-12 om 14 58 33" src="https://github.com/user-attachments/assets/25ee6d72-e7a2-4f06-948c-50dfa8fe37e5">


collapsed (not:expanded)

<img width="56" alt="Scherm­afbeelding 2024-11-12 om 14 58 39" src="https://github.com/user-attachments/assets/8bb8b6ae-4b0b-4c3f-9ad5-c30ddf448735">

I guess the fact there is no badge in the official notification tray is a Frontend bug.

**Reply by Mariusthvdb** [reply]
Wait, is this what you mean?

![IMG_5536](https://github.com/user-attachments/assets/c3775a4a-db81-483b-90fa-d5f0257d2389)

tbh I’ve never checked that as it only is a dot and holds no info.

also, I don’t have that on desktop, only in Companion


___edit___

I can force it to show when minimizing the window width:

<img width="356" alt="Scherm­afbeelding 2024-11-12 om 15 45 40" src="https://github.com/user-attachments/assets/003454e2-528e-47de-bcd5-7aa20331b078">

so the means my mod does not interfere with that Dot ;-)

btw I had accent-color set to primary color, so in that single spot made me overlook that easily.
I've set it to my alert-color now, and that makes it visible:
<img width="316" alt="Scherm­afbeelding 2024-11-12 om 16 11 36" src="https://github.com/user-attachments/assets/316f2767-1474-4154-bcfd-85cbea3cce2b">

unfortunately it make the notification badge too prominent for my taste (reason I had set it to primary-color) ...

<img width="251" alt="Scherm­afbeelding 2024-11-12 om 16 11 42" src="https://github.com/user-attachments/assets/1adff07f-8946-4af9-b942-7a8ff86a289d">

seems ok if you think about it as an alerter, but for me that is too distracting, having a badge there itself was enough for me

so, I figured, why not use your new theme variables and set those generically with:

```
    custom-sidebar-notification-color: var(--primary-color)
    custom-sidebar-notification-text-color: ivory
```

this way the notification always blends in, and, if must be, I can override them

<img width="295" alt="Scherm­afbeelding 2024-11-12 om 16 20 33" src="https://github.com/user-attachments/assets/0f77247b-d657-48c3-894c-326e1c22256f">

**Reply by elchininet** [reply]
>Uhm, what native notification is that? I’ve never seen anything there?

Because you never set the sidebar in that mode, that appears when the sidebar is in `always_hidden` mode. It happens when you enable `Always hide the sidebar` in your profile and it is the default mode in mobile devices. (you can also enable this mode if you set `sidebar_mode` as `hidden` with `custom-sidebar`)

>Wait, is this what you mean?

Yes

>also, I don’t have that on desktop, only in Companion
>I can force it to show when minimizing the window width:

In desktop it appears if you enable the mode that I mentioned before.

>so, I figured, why not use your new theme variables and set those generically with:
>
>    custom-sidebar-notification-color: var(--primary-color)
>    custom-sidebar-notification-text-color: ivory

You can use the plugin options, but those are not the options, the options are `notification_color ` and `notification_text_color`.

**Reply by Mariusthvdb** [reply]
Ive set those in my theme, not in the plugin.

But as I demonstrated, my card-mod theme mod for the notification on the menu icon does not interfere with the Dot, which was what you asked
